docs/workshops/interviews/technical/workshop.md

```diff
@@ -0,0 +1,262 @@
+function longestWord(sentence) {
+    const sentenceWithoutPunctuation = sentence.replace(/[^A-Za-z ]g/, ""); // "I like cats"
+    const words = sentenceWithoutPunctuation.split(" "); // ["I", "like", "cats"]
+    let longestWord = words[0]; // "I"
+    for (const word of words) { // word = "like"
+        if (word.length > longestWord.length) { // 4 > 1
+            longestWord = word;
+        }
+    }
+    return longestWord;
+}
+```
+
+#### Suggest possible refactorings/improvements
+
+If you think you'd like to change your code to improve it (e.g. to improve a variable name, or to extract a function for a piece of repeated code), describe how you would like to change it to your interviewer.
+
+Because writing code on whiteboards or shared text environments can be fiddly, the interviewer may just say "That's a good idea, but let's not bother", or they may ask you to make the change. But _you_ suggesting an improvement highlights that you care about your code quality.
+
+Of course, if you think your code is fine as it is, that's fine too!
+
+## Answering questions about your solution
+
+Your interviewer may now ask you some questions about your solution.
+
+Some examples of things the interviewer may ask about:
+* Why you chose to do something one way rather than another (e.g. why you used a `for` loop instead of a call to `map`).
+* Whether there are any interesting edge cases your code doesn't handle.
+* If there is anything you would change about the code.
+* TODO: Think about how to discuss: If there are any trade-offs you've made in your code
+* TODO: Think about whether to discuss efficiency at all.
+* TODO: Expand this list.
+
+:::tip
+Asking you about other ways of solving the problem does not mean you did it wrong! It can still be interesting to talk about other ways to do things, even if yours is a good one!
+:::
+
+## Handling changing constraints
+
+After you've finished solving the problem, your interviewer may ask you to change your code, or change the problem a bit. **This does not mean anything was wrong with your code**, it's a normal way to expand a problem.
+
+For example, with our example question, the interviewer may tell you that, actually, words containing punctuation should be completely ignored and must never be returned, or that punctuation inside words counts as a letter. The interviewer may ask you to change your code, or may just ask you to talk through how you would change it without actually doing so.
+
+When making this kind of change, follow exactly the same steps as you did the first time - make sure you understand the problem, talk through any edge cases, and remember to keep thinking aloud!
+
+Follow-up questions and changes **do not mean that you made a mistake**. They're a normal part of an interview - don't worry about them.
+
+## Workshop
+
+We're now going to split into breakout rooms and practice some interview questions together. Your TA will have a list of technical interview questions to try out.
```

Comment:
Could we add rubric or guide for how these could be assessed? It'd be cool to give the students something at the end of this that says 

"Your speaking out loud was 4/5 but you need to work on asking for help."

It needn't be very heavy touch, just something that gets our TAs in the mindset that they should be accessing to some degree.